### PR TITLE
[connectors] Investigation on Confluence usage

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Add a script that checks how many pages were modified in the last `n` minutes.

### CODE REVIEW

File: connectors/migrations/20241205_check_confluence_modified_pages.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+} from "@connectors/connectors/confluence/temporal/activities";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+
+makeScript(
+  {
+    timeWindowMs: {
+      type: "number",
+      demandOption: false,
+      default: 60 * 60 * 1000,
+      description: "Number of connectors to process concurrently",
```
Comment (flvndvd) [inline connectors/migrations/20241205_check_confluence_modified_pages.ts:16]: 🤔

File: connectors/migrations/20241205_check_confluence_modified_pages.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      const spaceIds = await getSpaceIdsToSyncActivity(connector.id);
+
+      for (const spaceId of spaceIds) {
+        const { pages } = await client.getPagesInSpace(
+          spaceId,
+          "all",
+          "-modified-date"
+        );
+        const recentlyModifiedPages = pages.filter(
+          (page) => new Date(page.version.createdAt) >= new Date(timeWindowMs)
```
Comment (flvndvd) [inline connectors/migrations/20241205_check_confluence_modified_pages.ts]: now - timeWindowMs ?

File: connectors/src/connectors/confluence/lib/confluence_client.ts [hunk 3]
```diff
@@ -454,12 +454,13 @@ export class ConfluenceClient {
   async getPagesInSpace(
     spaceId: string,
     depth: "all" | "root" = "all",
+    sort: "id" | "-modified-date" = "id",
```
Comment (flvndvd) [inline connectors/src/connectors/confluence/lib/confluence_client.ts:457]: You'll need to modify the call sites.

File: connectors/migrations/20241205_check_confluence_modified_pages.ts (hunk 2 above)
Comment (aubin-tchoi) [reply connectors/migrations/20241205_check_confluence_modified_pages.ts]: fixed

File: connectors/migrations/20241205_check_confluence_modified_pages.ts (hunk 1 above)
Comment (flvndvd) [reply connectors/migrations/20241205_check_confluence_modified_pages.ts:16]: Bump

File: connectors/migrations/20241205_check_confluence_modified_pages.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+      const client = await getConfluenceClient({
+        cloudId,
+        connectorId: connector.id,
+      });
+
+      const spaceIds = await getSpaceIdsToSyncActivity(connector.id);
+
+      for (const spaceId of spaceIds) {
+        const { pages } = await client.getPagesInSpace(
```
Comment (flvndvd) [inline connectors/migrations/20241205_check_confluence_modified_pages.ts:39]: So we only take the first 25 pages. If we want a full picture we might want to do a `do/while`, no?

Comment (aubin-tchoi) [reply connectors/migrations/20241205_check_confluence_modified_pages.ts]: done

File: connectors/migrations/20241205_check_confluence_modified_pages.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        cloudId,
+        connectorId: connector.id,
+      });
+
+      const spaceIds = await getSpaceIdsToSyncActivity(connector.id);
+
+      for (const spaceId of spaceIds) {
+        const allPages: ({
+          version: { number: number; createdAt: string };
+        } & { [p: string]: unknown })[] = [];
```
Comment (flvndvd) [inline connectors/migrations/20241205_check_confluence_modified_pages.ts:43]:
```suggestion
        const allPages: ReturnType<typeof getPagesInSpace["pages"]> = [];
```

**Review by flvndvd — APPROVED** [review]

**Pull request merged**